dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/registry/RegistryClient.java

```diff
@@ -0,0 +1,463 @@
+    public List<String> getServerNodeList(NodeType nodeType, boolean hostOnly) {
+        Set<String> serverSet = getServerNodeSet(nodeType, hostOnly);
+        List<String> serverList = new ArrayList<>(serverSet);
+        Collections.sort(serverList);
+        return serverList;
+    }
+
+    /**
+     * check the zookeeper node already exists
+     *
+     * @param host host
+     * @param nodeType zookeeper node type
+     * @return true if exists
+     */
+    public boolean checkNodeExists(String host, NodeType nodeType) {
+        String path = getNodeParentPath(nodeType);
+        if (StringUtils.isEmpty(path)) {
+            logger.error("check zk node exists error, host:{}, zk node type:{}",
+                    host, nodeType);
+            return false;
+        }
+        Map<String, String> serverMaps = getServerMaps(nodeType, true);
+        for (String hostKey : serverMaps.keySet()) {
+            if (hostKey.contains(host)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * @return get worker node parent path
+     */
+    protected String getWorkerNodeParentPath() {
+        return Constants.REGISTRY_DOLPHINSCHEDULER_WORKERS;
+    }
+
+    /**
+     * @return get master node parent path
+     */
+    protected String getMasterNodeParentPath() {
+        return Constants.REGISTRY_DOLPHINSCHEDULER_MASTERS;
+    }
+
+    /**
+     * @return get dead server node parent path
+     */
+    protected String getDeadNodeParentPath() {
+        return Constants.REGISTRY_DOLPHINSCHEDULER_DEAD_SERVERS;
+    }
+
+    /**
+     * @return get master lock path
+     */
+    public String getMasterLockPath() {
+        return Constants.REGISTRY_DOLPHINSCHEDULER_LOCK_MASTERS;
+    }
+
+    /**
+     * @param nodeType zookeeper node type
+     * @return get zookeeper node parent path
+     */
+    public String getNodeParentPath(NodeType nodeType) {
+        String path = "";
+        switch (nodeType) {
+            case MASTER:
+                return getMasterNodeParentPath();
+            case WORKER:
+                return getWorkerNodeParentPath();
+            case DEAD_SERVER:
+                return getDeadNodeParentPath();
+            default:
+                break;
+        }
+        return path;
+    }
+
+    /**
+     * @return get master start up lock path
+     */
+    public String getMasterStartUpLockPath() {
+        return Constants.REGISTRY_DOLPHINSCHEDULER_LOCK_FAILOVER_STARTUP_MASTERS;
+    }
+
+    /**
+     * @return get master failover lock path
+     */
+    public String getMasterFailoverLockPath() {
+        return Constants.REGISTRY_DOLPHINSCHEDULER_LOCK_FAILOVER_MASTERS;
+    }
+
+    /**
+     * @return get worker failover lock path
+     */
+    public String getWorkerFailoverLockPath() {
+        return Constants.REGISTRY_DOLPHINSCHEDULER_LOCK_FAILOVER_WORKERS;
+    }
+
+    /**
+     * init system node
+     */
+    public void initSystemNode() {
```

Comment:
The `initSystemNode`  method does the same thing with the parent method `RegistryCenter.initNodes()`.
We should better kept one.

And when execute persist method, it will delete all sub nodes. 
We should check whether the system node has been initialized.
 
```java
public void persist(String key, String value) {
        try {
            if (isExisted(key)) {
                client.delete().deletingChildrenIfNeeded().forPath(key);
            }
            client.create().creatingParentsIfNeeded().withMode(CreateMode.PERSISTENT).forPath(key, value.getBytes(StandardCharsets.UTF_8));

        } catch (Exception e) {
            throw new RegistryException("zookeeper persist error", e);
        }
    }
```